### PR TITLE
Misleading exception raised during batch scaling

### PR DESCRIPTION
**Pull request by tejasvi**
Use batch_size from `model.hparams.batch_size` instead of `model.batch_size`

### CODE REVIEW

**Comment by justusschock** [comment]
Hi @tejasvi . Thanks for contributing, but I probably wouldn't merge this, sice we recently moved away from hparams in #1896

**Comment by Borda** [comment]
@tejasvi mind update it to reflect we new structure of arguments?

**Comment by justusschock** [comment]
What about something like:

```python
splitted_batch_arg_name = batch_arg_name.split('.')
contains_arg = model
for name in splitted_batch_arg_name[:-1]:
    contains_arg = getattr(contains_arg, name)

batch_size = getattr(contains_arg, splitted_batch_arg_name[-1])

if value:
    setattr(contains_arg, splitted_batch_arg_name[-1], value)
```

This way one could either specify the name to something like `'batch_size'` (which would correspond to `model.batch_size`) or to `'hparams.batch_size'` (which would correspond to `model.hparamsbatch_size`).

The issue is, that arguments are not always attached and hparams can also be under other names (like `'hyperparameters'`). These issues would all be resolved with something like this I guess :)

**Comment by SkafteNicki** [comment]
We had a similar problem, when the learning rate finder was implemented, namely that some users like to store their learning rate in something like `model.hparams.optimizers.lr` instead of just `model.hparams.lr`. We solved it by introducing these two utility functions:
https://github.com/PyTorchLightning/pytorch-lightning/blob/c3cf33d1de7de3444e3a4a7d45bc613d8e01a7c0/pytorch_lightning/trainer/lr_finder.py#L485-L501
Which can be repurposed for this PR (move to `utilities.parsing`?). A similar `nested_getattr` needs to be implemented.

File: pytorch_lightning/trainer/training_tricks.py [hunk 1]
```diff
@@ -136,7 +136,9 @@ def scale_batch_size(self,
 
         """
         if not hasattr(model, batch_arg_name):
-            raise MisconfigurationException(f'Field {batch_arg_name} not found in `model.hparams`')
+            if not hasattr(model, 'hparams') or not hasattr(model.hparams, batch_arg_name):
```
Comment (Borda) [inline pytorch_lightning/trainer/training_tricks.py:139]: now `hparams` is property
```suggestion
            if not hasattr(model.hparams, batch_arg_name):
```

File: pytorch_lightning/trainer/training_tricks.py [hunk 2]
```diff
@@ -136,7 +136,9 @@ def scale_batch_size(self,
...
+                raise MisconfigurationException(f'Neither of `model.batch_size` and'
+                                                f' `model.hparams.batch_size` found.')
```
Comment (Borda) [inline pytorch_lightning/trainer/training_tricks.py:141]:
```suggestion
                raise MisconfigurationException(
                    'Neither of `model.batch_size` and `model.hparams.batch_size` found.'
                )
```
no need for `f` if there is no variable inside

File: pytorch_lightning/trainer/training_tricks.py [hunk 3]
```diff
@@ -245,17 +247,23 @@ def _adjust_batch_size(trainer,
 
     """
     model = trainer.get_model()
-    batch_size = getattr(model, batch_arg_name)
+    if hasattr(model, batch_arg_name):
+        batch_size = getattr(model, batch_arg_name)
```
Comment (Borda) [inline pytorch_lightning/trainer/training_tricks.py:251]: we are not registering in the model anymore...

**Comment by williamFalcon** [comment]
sorry, merged by accident. mind re-opening?
(merge has been reversed)

**Comment by Borda** [comment]
> sorry, merged by accident. mind re-opening?
> (merge has been reversed)

merged cannot be reopened, it is already in master
and reverting and immediate merge does not solve the issue much
@williamFalcon pls ping me next time...

**Comment by Borda** [comment]
@tejasvi mind open a new PR?
